### PR TITLE
Fix unnecessary painting in CoverArtDelegate

### PR DESCRIPTION
**Pull request by daschuer**
I have created this during my my first attempt to fix the priority inversion #11128 when scrolling with cover arts column enabled. This is a bit of a speed up, but does not fix the actual problem. This will be part of the next PR #13719

### CODE REVIEW

**Comment by ronso0** [comment]
FWIW I'm not noticing a difference in performance between 2.4 and this.
With both branches, for covers that are skipped while scrolling fast (only the color is painted) the images are loaded only when I scroll back to the skipped rows. Ie. I don't see any effect of c36e5661d8ddfea072f3d190a26db6383ddfca01/6c841bd4b2320f6dd4ebf0599c4f3d6697caf9e4 (IIUC how those are supposed to work).

**Review by ronso0 — CHANGES_REQUESTED** [review]
Just some small findings, I'll take a closer look soonish.

**Comment by daschuer** [comment]
The actual but small speed up happens once you stop scrolling. In this case the cover pictures appears faster. In the original version the uni-color replacements are first painted again after scroll stop and than the original covers are fetched and pained.

The other optimizations has only a negligible speed up. In 2.4 the chache grows up to 1000 and more, which requires repeated memory allocations. This branch reuses the memory once it is allocated for the number of all visible rows.

File: src/library/coverartdelegate.cpp [hunk 1]
```diff
@@ -54,20 +56,44 @@ void CoverArtDelegate::emitRowsChanged(
...
+        CoverArtCache::requestUncachedCover(this, coverInfo, width);
+    }
+    m_pendingCacheRows.insert(coverInfo.cacheKey(), row);
+}
+
 void CoverArtDelegate::slotInhibitLazyLoading(
         bool inhibitLazyLoading) {
     m_inhibitLazyLoading = inhibitLazyLoading;
-    if (m_inhibitLazyLoading || m_cacheMissRows.isEmpty()) {
+    if (m_inhibitLazyLoading) {
```
[the remark this answers is not in the document]
Comment (daschuer) [reply src/library/coverartdelegate.cpp:79]: I will restore it. It was no longer required in an intermediate version

**Comment by daschuer** [comment]
Done

File: src/library/coverartdelegate.cpp [hunk 2]
```diff
@@ -157,12 +158,31 @@ void CoverArtDelegate::paintItem(
...
     if (option.state & QStyle::State_HasFocus) {
         drawBorder(painter, m_pFocusBorderColor, option.rect);
     }
 }
+
+void CoverArtDelegate::cleanCacheMissRows() const {
+    auto it = m_cacheMissRows.begin();
+    while (it != m_cacheMissRows.end()) {
+        QModelIndex index = m_pTableView->model()->index(*it, m_column);
+        QRect rect = m_pTableView->visualRect(index);
```
[the remark this answers is not in the document]
Comment (ronso0) [reply src/library/coverartdelegate.cpp:179]: Only that one is open.

**Comment by daschuer** [comment]
Done

**Review by ronso0 — CHANGES_REQUESTED** [review]
LGTM, but let's polish that comment

**Comment by daschuer** [comment]
Done

**Comment by ronso0** [comment]
Thank you!

**Pull request merged**